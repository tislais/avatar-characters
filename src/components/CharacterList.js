import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const CharacterList = ({ characters }) => {
  
  const characterElements = characters.map((character) => (

    <li key={character._id} 
      className="p-5 bg-white rounded shadow-md 
      hover:scale-105 transition duration-300 ease-in-out
      cursor-pointer">

      <Character
        name={character.name}
        allies={character.allies}
        enemies={character.enemies}
        photoUrl={character.photoUrl}
        id={character._id}
      />

    </li>
  ))
  return (
    <ul 
      className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 2xl:grid-cols-6 p-4 gap-4">
      {characterElements}
    </ul>
  )
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    affiliation: PropTypes.string.isRequired,
    allies: PropTypes.array,
    enemies: PropTypes.array,
    photoUrl: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;