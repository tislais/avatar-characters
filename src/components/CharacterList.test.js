import React from 'react';
import { render } from '@testing-library/react';
import CharacterList from './CharacterList';

const characters = [...Array(5)].map((_, i) => ({
    name: "46th Earth King",
    allies: ["Royal Earthbender Guards"],
    enemies: ["Chin"],
    photo: "url"
  })
)

describe('Character component', () => {
  it('renders Character', () => {
    const { asFragment } = render(<CharacterList characters={characters} />);
    expect(asFragment()).toMatchSnapshot();
  });
});