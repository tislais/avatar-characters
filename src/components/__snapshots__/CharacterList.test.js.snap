// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <ul
    class="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 2xl:grid-cols-6 p-4 gap-4"
  >
    <li
      class="p-5 bg-white rounded shadow-md 
 hover:scale-105 transition duration-300 ease-in-out
 cursor-pointer"
    >
      <figure
        class=""
      >
        <img
          alt="46th Earth King"
          class="
 h-[300px] sm:h-[200px] 
 w-full 
 object-cover
 "
          src="url"
        />
        <figcaption>
          <p
            class="uppercase text-lg font-bold tracking-wider"
          >
            46th Earth King
          </p>
          <p>
            Allies: 
            <span>
              Royal Earthbender Guards
            </span>
          </p>
          <p>
            Enemies: 
            <span>
              Chin
            </span>
          </p>
        </figcaption>
      </figure>
    </li>
    <li
      class="p-5 bg-white rounded shadow-md 
 hover:scale-105 transition duration-300 ease-in-out
 cursor-pointer"
    >
      <figure
        class=""
      >
        <img
          alt="46th Earth King"
          class="
 h-[300px] sm:h-[200px] 
 w-full 
 object-cover
 "
          src="url"
        />
        <figcaption>
          <p
            class="uppercase text-lg font-bold tracking-wider"
          >
            46th Earth King
          </p>
          <p>
            Allies: 
            <span>
              Royal Earthbender Guards
            </span>
          </p>
          <p>
            Enemies: 
            <span>
              Chin
            </span>
          </p>
        </figcaption>
      </figure>
    </li>
    <li
      class="p-5 bg-white rounded shadow-md 
 hover:scale-105 transition duration-300 ease-in-out
 cursor-pointer"
    >
      <figure
        class=""
      >
        <img
          alt="46th Earth King"
          class="
 h-[300px] sm:h-[200px] 
 w-full 
 object-cover
 "
          src="url"
        />
        <figcaption>
          <p
            class="uppercase text-lg font-bold tracking-wider"
          >
            46th Earth King
          </p>
          <p>
            Allies: 
            <span>
              Royal Earthbender Guards
            </span>
          </p>
          <p>
            Enemies: 
            <span>
              Chin
            </span>
          </p>
        </figcaption>
      </figure>
    </li>
    <li
      class="p-5 bg-white rounded shadow-md 
 hover:scale-105 transition duration-300 ease-in-out
 cursor-pointer"
    >
      <figure
        class=""
      >
        <img
          alt="46th Earth King"
          class="
 h-[300px] sm:h-[200px] 
 w-full 
 object-cover
 "
          src="url"
        />
        <figcaption>
          <p
            class="uppercase text-lg font-bold tracking-wider"
          >
            46th Earth King
          </p>
          <p>
            Allies: 
            <span>
              Royal Earthbender Guards
            </span>
          </p>
          <p>
            Enemies: 
            <span>
              Chin
            </span>
          </p>
        </figcaption>
      </figure>
    </li>
    <li
      class="p-5 bg-white rounded shadow-md 
 hover:scale-105 transition duration-300 ease-in-out
 cursor-pointer"
    >
      <figure
        class=""
      >
        <img
          alt="46th Earth King"
          class="
 h-[300px] sm:h-[200px] 
 w-full 
 object-cover
 "
          src="url"
        />
        <figcaption>
          <p
            class="uppercase text-lg font-bold tracking-wider"
          >
            46th Earth King
          </p>
          <p>
            Allies: 
            <span>
              Royal Earthbender Guards
            </span>
          </p>
          <p>
            Enemies: 
            <span>
              Chin
            </span>
          </p>
        </figcaption>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
